src/scipp/core/data_group.py

```diff
@@ -198,12 +198,15 @@ def _repr_html_(self):
         return out
 
     def __repr__(self):
-        r = 'DataGroup(\n'
+        r = f'DataGroup(sizes={self.sizes}, keys=[\n'
         for name, var in self.items():
-            r += f'    {name}: {_summarize(var)}\n'
-        r += ')'
+            r += f'    {name}: {type(var).__name__},\n'
```

Comment:
Can we keep calling `_summarize` here? Or something similar. It would be nice to at least see the dims and shape of entries.

Reply:
In the future, we might want to add more types to summarize. Maybe this would even be useful in other parts of the code, e.g. to print coords. So maybe we will generalise it.
But for now, this solution looks good to me.